moveos/moveos-stdlib/moveos-stdlib/sources/object.move

```diff
@@ -22,24 +26,43 @@ module moveos_std::object {
     /// The object can not be copied, droped, only can be consumed by `add`
     struct Object<T> {
         id: ObjectID,
-        owner: address,
+        //owner: address,
+        owner: Option<Owner>,
         //TODO define shared and immutable
         //shared: bool,
         //immutable: bool,
         // The value must be the last field
         value: T,
     }
 
+    public(friend) fun set_address_owner<T>(obj: &mut Object<T>, owner: address) {
```

Comment:
verification is required when setting owner, for example, transfer is not allowed after freeze and share